src/mon/OSDMonitor.cc

```diff
@@ -12480,17 +12485,29 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       goto wait;
     if (err < 0)
       goto reply_no_propose;
+
     pg_t pgid;
-    err = parse_pgid(cmdmap, ss, pgid);
-    if (err < 0)
-      goto reply_no_propose;
-    if (pending_inc.old_pools.count(pgid.pool())) {
-      ss << "pool of " << pgid << " is pending removal";
-      err = -ENOENT;
-      getline(ss, rs);
-      wait_for_commit(op,
-        new Monitor::C_Command(mon, op, err, rs, get_last_committed() + 1));
-      return true;
+    switch (upmap_option) {
+    case OP_PG_UPMAP:
+    case OP_RM_PG_UPMAP:
+    case OP_PG_UPMAP_ITEMS:
+    case OP_RM_PG_UPMAP_ITEMS:
+    case OP_PG_UPMAP_PRIMARY:
+    case OP_RM_PG_UPMAP_PRIMARY:
+      err = parse_pgid(cmdmap, ss, pgid);
+      if (err < 0)
+	goto reply_no_propose;
+      if (pending_inc.old_pools.count(pgid.pool())) {
+	ss << "pool of " << pgid << " is pending removal";
+	err = -ENOENT;
+	getline(ss, rs);
+	wait_for_commit(op,
+	  new Monitor::C_Command(mon, op, err, rs, get_last_committed() + 1));
+	return true;
+      }
+      break;
+    case OP_RM_PG_UPMAP_PRIMARY_ALL: // no pgid to check
```

Comment:
very nit ;) - I would put this case as the first one - else it is hard to find it.
I also like to add `default:` case with an error message or assert just in case someone adds a value in the future and forgets to update all the places

Reply:
ACK. In the interest of time (this fix is for v18.2.5), I will add this in a followup part 2. The compiler does warn when you do not include a case, so I think that can be a small cleanup that does not have a major impact on the logic.

Reply:
ACK, added. This has already gone into teuthology testing, although the change is so trivial that I will confirm locally that the fix works, then IMO it is okay to merge pending QA approval.

Reply:
Confirmed locally.